src/Hosting/Hosting/src/Host.cs

```diff
@@ -71,8 +72,11 @@ public static IHostBuilder CreateDefaultBuilder(string[] args)
             {
                 var env = hostingContext.HostingEnvironment;
 
-                config.AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
-                      .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true, reloadOnChange: true);
+                var reloadOnChange = Convert.ToBoolean(
+                    Environment.GetEnvironmentVariable("ASPNETCORE_BUILDER_CONFIG_RELOAD") ?? "true");
```

Comment:
Rather than directly reading an environment variable, how about reading from the host's config (which does include environment variables already, but could also be configured other ways). See hostingContext.Configuration. https://github.com/dotnet/extensions/blob/24751691256010d4fac8d16afc56e6d9db12c00a/src/Hosting/Abstractions/src/HostBuilderContext.cs#L27
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-3.1#configure-the-host-builder-with-configurehostconfiguration

Reply:
Oh cool, I was worried about config not being 'ripe' by this point, but that does make sense since it's pulled in already, thanks.

Pushing an update that
* Switches option pull from `Environment` to `hostingContext.Configuration`
* Changes flag name to `DOTNET_HOSTBUILDER_CONFIG_RELOAD` to disambiguate and fit the [config.AddEnvironmentVariables prefix](https://github.com/dotnet/extensions/blob/master/src/Hosting/Hosting/src/Host.cs#L63)
* Update tests to use `DOTNET_HOSTBUILDER_CONFIG_RELOAD`